### ISSUE TITLE
Isn't the double.infinity code mathematically Incorrect?

### ISSUE THREAD

**Issue by ThinkDigitalRepair** [dart-lang/sdk#34600]
The current Dart code shows the following for the NaN and infinity variables.
https://github.com/dart-lang/sdk/blob/7c54fab8a5736343aa6bef1f45c2e1c5152a94b4/sdk/lib/core/double.dart#L26-L27

According to... standard math and computers, dividing a non-zero number by 0 is impossible. It results in an error because it does not compute. So to have `double.infinity` be _computed_ by using an _incomputable_ equation makes no sense. It would make more sense if `double.nan` was computed using 1.0/0.0.

According to [Ask Dr. Math](http://mathforum.org/dr.math/faq/faq.divideby0.html) , or any other site for that matter, dividing 0 by 0 results in an indeterminate number, which would be a close value to infinity than an equation that will fail to calculate.

So I propose a simple switch in the calculating of these values just for making semantic sense.

```dart
static const double nan = 1.0 / 0.0;
static const double infinity = 0.0 / 0.0;
```
Dart VM version: 2.0.0-dev.61.0 (Mon Jun 11 15:46:09 2018 +0200) on "macos_x64"

**Comment by lrhn** [comment]
In short: This is not math, it's engineering.

Dart `double` values are 64-bit [IEEE-754](https://en.wikipedia.org/wiki/IEEE_754) floating point numbers, and their behavior is governed by that standard (which is, not coincidentally, also the standard implemented by all CPUs).

Such floating point numbers represent inexact values. There is only a finite number of double values, but together they represent all real numbers because each double represents a range. The double for 1.5 really represents the range of real numbers centered around 1.5 going half way in each direction to the adjacent representable doubles. In a sence, double numbers are really "numbers with error-bars", which is where the engineering comes in (it's also the last "E" in IEEE).

The values for zero and infinity represent the caps of those ranges, underflow and overflow, and not any exact value. The infinity value just means "larger than we can represent, nobody knows by how much" and zero means "smaller than we can represent, nobody knows how small".

The IEEE_754 specification states that a non-zero number divided by zero becomes infinity. That makes sense if you see zero as representing some arbitrarily small number, and infinity as representing an arbitrarily large number. Any known value divided by something arbitrarily small can get arbitrarily big.

The specification also says that 0.0 / 0.0 is *undefined* (so is infinity / infinity) because the result can really be anywhere in the real numbers. The undefined result is represented as the Not-a-Number value.

So, in summary, the ways we compute the double NaN and infinity values are not something we have come up with, it's just how doubles work, in Dart *and* in any other language with similar floating point numbers, because it's what we can efficiently implement on any modern CPU.

**Comment by ThinkDigitalRepair** [comment]
Thank you for taking the time to explain this!
